hibernate-rx-core/src/main/java/org/hibernate/rx/engine/impl/RxPersistenceContext.java

```diff
@@ -0,0 +1,511 @@
+package org.hibernate.rx.engine.impl;
+
+import org.hibernate.HibernateException;
+import org.hibernate.LockMode;
+import org.hibernate.MappingException;
+import org.hibernate.collection.spi.PersistentCollection;
+import org.hibernate.engine.loading.internal.LoadContexts;
+import org.hibernate.engine.spi.*;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.persister.entity.EntityPersister;
+import org.hibernate.rx.engine.loading.impl.RxLoadContexts;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+public class RxPersistenceContext implements PersistenceContext {
```

Comment:
@DavideD What is the purpose of this class? It has no Javadoc? Am I right in surmising it exists just to plug in an `RxLoadContexts`? Is there a good reason to not just extend the public class `StatefulPersistenceContext`?

Reply:
1. I don't like to extend if I can avoid it
2. This way we could potentially reuse this for other `PersistenceContext`s

Reply:
Sorry, I've just realized it's not needed in this PR.

Reply:
> 1. I don't like to extend if I can avoid it

Well by `extend`ing a class and overriding just one method, instead of implementing an interface and overriding tens of methods, we limit our exposure to changes in hibernate-core. We don't want to have hibernate-rx break every time there is a totally unrelated change to a method signature in core.

> 2. This way we could potentially reuse this for other `PersistenceContext`s

Are there other implementations that are relevant to Hibernate RX?

> Sorry, I've just realized it's not needed in this PR.

Even better.

Reply:
> Well by extending a class and overriding just one method, instead of implementing an interface and overriding tens of methods, we limit our exposure to changes in hibernate-core. 

Note that there's an ancient bit of folklore from back in the 90s that holds that class extension represents (supposedly bad) "implementation inheritance" whereas interface implementation being pure type inheritance is something that is somehow more maintainable. This was always mostly nonsense, since adding an abstract operation to an interface is a change that breaks implementations of the interface. This bit of folklore was implicitly rejected by the gods of Java a few years ago by when support for `default` interface methods was added. But most of us had figured out that it was rubbish a bit before then.

Reply:
In this case it made sense to me to add a delegator because I didn't really care about the specific implementation of the interface, PersistenceContext doesn't change that often and when it does it's easy to notice it and update the delegator. In practice,SatefulPersistenceContext seems to be the only implementation so I could have extended it.